File: compositor/src/main/java/org/westford/compositor/core/events/RenderOutputNew.java
package org.westford.compositor.core.events;

import com.google.auto.value.AutoValue;
import org.westford.compositor.protocol.WlOutput;

import javax.annotation.Nonnull;

@AutoValue
public abstract class RenderOutputNew {
    public static RenderOutputNew create(@Nonnull final WlOutput wlOutput) {
        return new AutoValue_RenderOutputNew(wlOutput);
    }

    @Nonnull
    public abstract WlOutput getWlOutput();
}

File: compositor/src/main/java/org/westford/compositor/core/events/RenderOutputNew.kt
package org.westford.compositor.core.events

import com.google.auto.value.AutoValue
import org.westford.compositor.protocol.WlOutput

@AutoValue
abstract class RenderOutputNew {

    abstract val wlOutput: WlOutput

    companion object {
        fun create(wlOutput: WlOutput): RenderOutputNew {
            return AutoValue_RenderOutputNew(wlOutput)
        }
    }
}
